src/wasm.h

```diff
@@ -176,49 +176,60 @@ struct Literal {
     }
   }
 
+  int32_t floatPayload() {
+    assert(type == WasmType::f32);
+    assert(std::isnan(f32));
+    return ~0xffc00000u & bit_cast<uint32_t>(f32);
```

Comment:
The payload mask should be 0xff800000 as wasm considers the quiet-nan bit to be part of the payload.